HW 02/Tast 2.2/Matrix.cs

```diff
@@ -0,0 +1,123 @@
+﻿namespace Tast_2._2
+{
+    public class Matrix
+    {
+        private int size;
+        private int[] array;
+
+        public Matrix(params int[] diagonal)
+        {
+            if (diagonal != null)
+            {
```

Comment:
.A list of parameters (params) is passed to the constructor - these are matrix elements located on the diagonal. The size of _array has to be  diagonal.Length, not size*size. And copy diagonal to _array, no need to set out of diagonal elements to 0, this behavior will be simulated by indexer ( GenerateMatrix is an unnesessary method)